strawberryfields/io.py

```diff
@@ -0,0 +1,214 @@
+                op["kwargs"]["select"] = cmd.op.select
+
+            if cmd.op.p:
+                # argument is quadrature phase
+                op["kwargs"]["phi"] = cmd.op.p[0].x
+
+        else:
+            if cmd.op.p is not None:
+                for a in cmd.op.p:
+                    # check if reg ref transform
+                    if isinstance(a.x, ops.RegRefTransform):
+                        if a.x.func_str is not None:
+                            # Note: not currently guaranteed to work,
+                            # as the RegRefTransform string cannot be checked
+                            # to determine if it is a valid function for serialization!
+                            #
+                            # Possible solutions:
+                            #
+                            #   * Use SymPy to represent functions, as
+                            #     SymPy provides methods for converting to
+                            #     Python functions as well as serialization
+                            #
+                            #   * Don't allow classical processing of measurements
+                            #     on remote backends
+                            #
+                            op["args"].append(a.x.func_str)
+                        else:
+                            raise ValueError(
+                                "The RegRefTransform in operation {} "
+                                "is not supported by Blackbird.".format(cmd.op)
+                            )
+                    else:
+                        op["args"].append(a.x)
+
+        bb._operations.append(op)
+
+    return bb
+
+
+def to_program(bb):
+    """Convert a Blackbird Program to a Strawberry Fields Program.
+
+    Args:
+        bb (blackbird.BlackbirdProgram): the input Blackbird program object
+
+    Returns:
+        Program: corresponding SF program
+    """
+    # create a SF program
+    if not bb.modes:
+        # we can't return an empty program, since we don't know how many modes
+        # to initialize the Program object with.
+        raise ValueError("Blackbird program contains no quantum operations!")
+
+    prog = Program(max(bb.modes)+1, name=bb.name)
+
+    # append the quantum operations
+    with prog.context as q:
+        for op in bb.operations:
+            try:
+                if 'args' in op:
+                    getattr(ops, op["op"])(*op["args"], **op["kwargs"]) | [q[i] for i in op["modes"]] #pylint:disable=expression-not-assigned
+                else:
+                    getattr(ops, op["op"]) | [q[i] for i in op["modes"]] #pylint:disable=expression-not-assigned
+            except AttributeError:
+                raise NameError("Quantum operation {} not defined!".format(op["op"]))
+
+    # compile the program if a target exists
+    if bb.target["name"] is not None:
+        return prog.compile(bb.target["name"])
+
+    return prog
+
+
+def save(file, prog):
```

Comment:
Github is highlighting `file`. I checked and `file` seems to be a builtin in Python. However, it looks like it is a python2 thing: https://stackoverflow.com/questions/24942358/is-file-a-keyword-in-python,
so I guess it's ok to use `file` instead of `file_`?

Reply:
Oops. My syntax highlighter doesn't highlight it. Okay to leave it as is?

Reply:
![image](https://user-images.githubusercontent.com/2959003/56776219-1155be80-6799-11e9-9544-8b69b9d94644.png)